### ISSUE TITLE
Porblem setting C# Decimal values

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#36707]
When setting a &quot;decimal&quot; value and writing it to the Console like so:

decimal negexp = 9.012345678901234567890123456e-26m;
Console.WriteLine($&quot;{test:E28}&quot;);

I get:
9.0100000000000000000000000000E-026

but SHOULD be: 9.0123456789012345678901234560E-026.

I also tried smaller (negative) exponents:

negexp = 9.012345678901234567890123456e-16m;
Console.WriteLine($&quot;{test:E28}&quot;);

and got:
9.0123456789000000000000000000E-016

and:

negexp = 9.012345678901234567890123456e-6m;
Console.WriteLine($&quot;{test:E28}&quot;);

and got:
0.0000090123456789012345678901

and:

negexp = 9.012345678901234567890123456e-2m;
Console.WriteLine($&quot;{test:E28}&quot;);

and got:
0.0901234567890123456789012346.

I also tried it with a positive exponent:

decimal posexp = 9.012345678901234567890123456e+26m;
Console.WriteLine($&quot;{test:E28}&quot;);

and got:
9.0123456789012345678901234560E+026

which is correct. 

So the acccuracy lost depends on the magnitude of the (negative) exponent and appears to be a pretty serious bug (unless I'm doing something wrong). I believe that these numbers are well within the advertised accuracy of the &quot;decimal&quot; data type according to the documentation.

I found it hard to believe that nobody has noticed this yet, but I DID do a search with no luck. Sorry if I missed it somewhere.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/506155/porblem-setting-c-decimal-values.html
VSTS ticketId: 828612_
_These are the original issue comments:_

Visual Studio Feedback System on 3/27/2019, 01:18 AM (89 days ago): <p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by gafter** [comment]
This looks correct.  A decimal value can have digits of precision anywhere from `1E-28` to `1E28`.  For the latter number following digits of precision are within that range, but for the value `1E-28` following digits are out of range of the data type and cannot contribute precision.

The smallest `decimal` value is `1.0E-28`.  For a value at the small end of the range, no additional digits beyond that value are possible.  Your number `9.012345678901234567890123456e-26m` has only the digits of precsion `9.01E-26`, as the following digits are less than `1.0E-28` and not representable.

In other words, this is by design.
